bill/invoice.go

```diff
@@ -251,30 +253,63 @@ func (inv *Invoice) RemoveIncludedTaxes(accuracy uint32) *Invoice {
 	return &i2
 }
 
-func (inv *Invoice) prepareSchemes(r *tax.Regime) error {
+// ScenarioSummary determines a summary of the tax scenario for the invoice based on
+// the document type and tax tags.
+func (inv *Invoice) ScenarioSummary() *tax.ScenarioSummary {
```

Comment:
I'm a fan of how this provides a much clearer/cleaner way for the client to obtain additional information than the previous iteration. I imagine it will be used like:

```go
scenarioSummary := inv.ScenarioSummary()

DatiGeneraliDocumento{
    TipoDocumento: scenarioSummary.Meta[it.KeyFatturaPATipoDocumento]
}
```

My immediate thought was it would also be helpful to have `Line.ScenarioSummary` so we can use the same technique to gain access to line-level codes for the case of Italy.

Reply:
Okay, so in the end I moved away from the Summaries as they technically add quite a bit of noise... the tags are at least validated on a per-line basis, but the provider is responsible for making the lookup and mapping. (We might need to review this.)